Kitodo/src/main/java/org/kitodo/production/helper/tasks/HierarchyMigrationTask.java

```diff
@@ -242,6 +247,33 @@ private static void migrateMetadataFiles(Process process) throws IOException {
         return parentData;
     }
 
+    private void checkTaskAndId(Process parentProcess) throws IOException {
+        URI parentMetadataFilePath = fileService.getMetadataFilePath(parentProcess, true, true);
+        Workpiece workpiece = ServiceManager.getMetsService().loadWorkpiece(parentMetadataFilePath);
+        ImportService.checkTasks(parentProcess, workpiece.getRootElement().getType());
```

Comment:
If the method `checkTasks` is required here as well (and not just during import), wouldn't it make sense to move that method from `ImportService` to `WorkflowControllerService` or `ProcessService`?